### DISCUSSION TITLE
https://bxb100.github.io/posts/tailscale_%E5%92%8C_surge_%E5%86%B2%E7%AA%81%E7%9A%84%E8%A7%A3%E5%86%B3%E6%96%B9%E6%A1%88/

### DISCUSSION THREAD

**Discussion by giscus** [bxb100/blog#62] [Announcements]
# https://bxb100.github.io/posts/tailscale_%E5%92%8C_surge_%E5%86%B2%E7%AA%81%E7%9A%84%E8%A7%A3%E5%86%B3%E6%96%B9%E6%A1%88/

John's Blog.

https://bxb100.github.io/posts/tailscale_%E5%92%8C_surge_%E5%86%B2%E7%AA%81%E7%9A%84%E8%A7%A3%E5%86%B3%E6%96%B9%E6%A1%88/

**Comment by wclebb** [comment]
感谢，原来 Surge 已经跟 Tailscale 兼容在一起了。这段时间还苦于怎么在保留 Surge 的情况下内网穿透。因为之前知晓过 Tailscale 是不能跟 Surge 兼容。

我刚考虑是否取消增强模式来让出给 Tailscale ，就搜了网上的是否有相关的例子，就看到了你。
第一行字足够让我兴奋，尝试安装果然能用。

内网穿透得好幸福。

**Reply by bxb100** [reply]
我感觉之前的理解和猜测并不靠谱, 可以看看新的 userspace-networking 模式, 这个确定性比较强, 也好理解

**Comment by llity** [comment]
我这里不兼容呢，打开tailscale，surge就无法代理

**Reply by bxb100** [reply]
大概率是你开了那啥 magic DNS, tailscale 写入了 nameserver, 可以 `cat /etc/resolv.conf` 看看是不是有 100.x.x.x

建议直接用 userspace-networking, 确定性比之前猜来猜去靠谱多了

**Comment by wclebb** [comment]
或许最新才能用。

Loong ***@***.***> 于2025年9月26日周五 19:20写道：

**Comment by s-leeq** [comment]
大佬，请教一下你有测试过Surge新的Tailscale代理吗？我无论怎么配置都会提示代理超时，给它配置规则也走不通，但是在Tailscale后台又能看到新的设备是Connected状态。

**Reply by bxb100** [reply]
没有，我用的 5.x
